packages/core/src/PartialExecutionUtils/findSubgraph.ts

```diff
@@ -87,15 +100,38 @@ function findSubgraphRecursive(
  *     - take every incoming connection and connect it to every node that is
  *       connected to the current node’s first output
  *   6. Recurse on each parent
+ *   7. Re-add all connections that don't use the `Main` connections type.
```

Comment:
Was it easier to implement it this way? I.e. removing them and adding them back later on, instead of keeping them in the graph during the traversal?

Reply:
I tried that first and it turned out really messy. 

I'd have to complicate `2. if the current node has no parents, don’t keep this branch` to `2. if the current node has no parents, don’t keep this branch, unless any connection on the branch has another type than Main` and that made the code harder to understand. I don't remember if that was just me getting confused or if it really was more difficult. 

I can try it again though. The good thing is the tests don't care how this works.